app/controllers/teachers_controller.rb

```diff
@@ -0,0 +1,67 @@
+class TeachersController < ApplicationController
+  before_action :authenticate_user!, :set_user
+
+  def show
+    @groups = @user.groups.all
+    @total_of_students = @user.group_members.count
+    # TODO: Add days of the weey to this query
+    @in_progress_group = @user.groups
+                              .where("from_time <= ?", Time.new)
+                              .where("to_time >= ?", Time.new).last
+  end
+
+  def show_students
+    @students = User.find(
+      [@user.group_members.map(&:user_id)]
+    )
+    render "teachers/students/index"
+  end
+
+  def new
+    @new_user = User.new
+  end
+
+  def edit; end
+
+  def create
+    @new_user = User.new(user_params)
+    # TODO: Add support to create students user
+    temp_password = @new_user.birth_date.strftime('%d%m%y')
```

Comment:
why don't we add this temp_password into the `user_params` above? so we have something like

`@new_user = User.new(user_params.merge(password: params[:birth_date].strftime('%d%m%y')))`

Reply:
Could be, but the purpose of that variable it's simplify the code because `params[:birth_date].strftime('%d%m%y')` can be confuse and that value is required in many lines, so, I propose a private function that return that value like this:
`@new_user = User.new(user_params.merge(password: new_user_temp_password))` where new_user_temp_password is the private function